webapp/src/pages/eventConfig/components/EventConfigComponent.js

```diff
@@ -0,0 +1,418 @@
+import React, { Component } from "react";
+import { eventService } from "../../../services/events";
+import { withRouter } from "react-router";
+import DateTimePicker from "react-datetime-picker";
+
+class EventConfigComponent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      preEvent: this.emptyEvent,
+      updatedEvent: this.emptyEvent,
+      hasBeenUpdated: false,
+      loading: true,
+      error: ""
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.event) {
+      eventService.getEvent(this.props.event.id).then(result => {
+        this.setState({
+          loading: false,
+          preEvent: result.event,
+          updatedEvent: result.event,
+          hasBeenUpdated: false,
+          error: result.error
+        });
+      });
+    }
+  }
+
+  onClickCancel = () => {
+    this.setState({
+      updatedEvent: this.state.preEvent,
+      hasBeenUpdated: false
+    });
+  };
+
+  onClickSubmit = () => {
+    // PUT
+    eventService.update(this.state.updatedEvent).then(result => {
+      this.setState({
+        preEvent: result.event,
+        updatedEvent: result.event,
+        hasBeenUpdated: false,
+        successButtonText: "Update Event",
+        updatingEvent: true,
+        addingNewEvent: false,
+        error: result.error
+      });
+    });
+  };
+
+  hasBeenEdited = () => {
+    let isEdited = false;
+    for (var propname in this.state.preEvent) {
+      if (this.state.updatedEvent[propname] !== this.state.preEvent[propname]) {
+        isEdited = true;
+      }
+    }
+    this.setState({
+      hasBeenUpdated: isEdited
+    });
+  };
+
+  updateEventDetails = (fieldName, e) => {
+    let u = { ...this.state.updatedEvent, [fieldName]: e.target.value };
+
+    this.setState({
+      updatedEvent: u
+    });
+    this.hasBeenEdited();
```

Comment:
You probably want to call this in the callback of setState, because at the moment there is no guarantee that the state would have updated before hasBeenEdited runs.

Reply:
Thanks, I didn't know about this.